src/tests/modify.test.js

```diff
@@ -0,0 +1,661 @@
+import { modify } from '../modify';
+
+/**
+ * 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧
+ * WIP — In progress
+ * Please review all tests explaining each spec.
+ * Tests with it.skip() are still not implemented.
+ * Thank you :)
+ * 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧 🚧
+ */
+
+const schemaPet = {
+  type: 'object',
+  additionalProperties: false,
+  properties: {
+    has_pet: {
+      title: 'Has Pet',
+      description: 'Do you have a pet?',
+      oneOf: [
+        {
+          title: 'Yes',
+          const: 'yes',
+        },
+        {
+          title: 'No',
+          const: 'no',
+        },
+      ],
+      'x-jsf-presentation': {
+        inputType: 'radio',
+      },
+      type: 'string',
+    },
+    pet_name: {
+      title: "Pet's name",
+      description: "What's your pet's name?",
+      'x-jsf-presentation': {
+        inputType: 'text',
+      },
+      type: 'string',
+    },
+    pet_age: {
+      title: "Pet's age in months",
+      maximum: 24,
+      'x-jsf-presentation': {
+        inputType: 'number',
+      },
+      'x-jsf-errorMessage': {
+        maximum: 'Your pet cannot be older than 24 months.',
+      },
+      type: 'integer',
+    },
+    pet_fat: {
+      title: 'Pet fat percentage',
+      'x-jsf-presentation': {
+        inputType: 'number',
+        percentage: true,
+      },
+      type: 'integer',
+    },
+    pet_address: {
+      properties: {
+        street: {
+          title: 'Street',
+        },
+      },
+    },
+  },
+  required: ['has_pet'],
+  'x-jsf-order': ['has_pet', 'pet_name'],
+  allOf: [
+    {
+      id: 'pet_conditional_id',
+      if: {
+        properties: {
+          has_pet: {
+            const: 'yes',
+          },
+        },
+        required: ['has_pet'],
+      },
+      then: {
+        required: ['pet_name', 'pet_age', 'pet_fat'],
+      },
+      else: {
+        properties: {
+          pet_name: false,
+          pet_age: false,
+          pet_fat: false,
+        },
+      },
+    },
+  ],
+};
+
+const schemaAddress = {
+  properties: {
+    address: {
+      properties: {
+        street: {
+          title: 'Street',
+        },
+        number: {
+          title: 'Number',
+        },
+        city: {
+          title: 'City',
+        },
+      },
+    },
+  },
+};
+
+const imagineSomeBasicSchema = {
+  /* ... */
+};
+
+/*
+modify() Options:
+
+{
+  fields,
+  allFields,
+  order,
+  conditionals,
+  pick,
+  omit,
+  split, 
+}
+
+*/
+
+describe('modify() - attributes', () => {
+  it('replace base field', () => {
+    const result = modify(schemaPet, {
+      fields: {
+        pet_name: {
+          title: 'Your pet name',
+        },
+        // Q:❓[*1] Do you find it useful to be a callback function instead?
+        // You can access any raw attribute from the field to do whatever verification you need,
+        // but remember to be cautious, as the attrs value might change.
+        has_pet: (fieldAttrs) => {
```

Comment:
Good question! We are not planning on that because this mutation happens _before_ initializing the form. It's called once and that's it. The same way a traditional JSON Schema does not have access to values.

Could you give an example on why you need it (simplified without disclosure business logic)?


edit: After seeing your next comment about React components, I know understand what you mean. So you see this modify being used not just as "mount" phase but for the entire "lifecycle" of a form (when filling the fields). Is that it?

Reply:
> Could you give an example on why you need it (simplified without disclosure business logic)?

We introduced a new logic in our codebase where we render some fields depending on the values of other fields. If we do not render a field we statically fill the value of the removed field. This is why I am asking that. The field we were hiding is the `available_pto` field in the Philippines schema. If the `available_pto_type` is selected is `unlimited` we hide the field rather than disabling it.